packages/react-draggable-dialog/src/components/DraggableDialog/DraggableDialog.test.tsx

```diff
@@ -1,16 +1,183 @@
 import * as React from 'react';
-import { render } from '@testing-library/react';
+import { act, render, renderHook } from '@testing-library/react';
+
 import { DraggableDialog } from './DraggableDialog';
+import { useDraggableDialog } from './useDraggableDialog';
+import { DragEndEvent } from '@dnd-kit/core';
+
+jest.mock('@fluentui/react-components', () => ({
+  ...jest.requireActual('@fluentui/react-components'),
+  useId: jest.fn().mockReturnValue('draggable-dialog'),
+}));
+
+const dialogChild = React.createElement('div', null, 'Dialog Child');
 
 describe('DraggableDialog', () => {
+  afterAll(() => {
+    jest.resetAllMocks();
+  });
+
   it('should render', () => {
     const text = 'Context';
     const { getByText } = render(
-      <DraggableDialog>
+      <DraggableDialog open>
         <div>{text}</div>
       </DraggableDialog>
     );
 
     expect(() => getByText(text)).toBeTruthy();
   });
+
+  it('should return default values', () => {
```

Comment:
wondering why are we testing internal hook ( not part of public api ) instead testing the component ?

Reply:
This component only renders child elements and holds state as Context to provide data down the tree. Since we separate render, state and style into files, the idea of testing the state hook here is to make sure we have correct state passed down as well as reacting to prop changes in the component (since the hook is tied with props).
What do you think about it? Do we have any other approach I can apply here?

Reply:
if you really need to test the implementation details ( the hook in this case ), I'd move it to related file to the hook implementation -> `useDraggableDialog.test.ts` 

I'd expect DraggableDialog.test to contain test for DraggableDialog component behaviours, not internal hook tests if that makes sense